### ISSUE TITLE
ECDSA key cannot be loaded / is not sent to server

### ISSUE THREAD

**Issue by p-mongo** [jruby/jruby-openssl#213]
When I use ECDSA keys on the client side, these keys are either not read or not sent to the server.

Test code + key/cert: https://github.com/p-mongo/tests/tree/master/jruby-ecdsa

## Test 1: PKey::EC.new on a pem file contents containing cert + private key

```
require 'openssl'
require 'socket'

tcp_socket = TCPSocket.new 'localhost', 9999
context = OpenSSL::SSL::SSLContext.new
context.cert = OpenSSL::X509::Certificate.new(File.read('server.pem'))
context.key = OpenSSL::PKey::EC.new(File.read('server.pem'))
ssl_socket = OpenSSL::SSL::SSLSocket.new tcp_socket, context
```

This key is successfully usable under MRI. JRuby 9.2.13.0 produces:

```
serene% ruby test.rb            
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/home/w/.rbenv/versions/jruby-9.2.13.0/lib/ruby/stdlib/jopenssl.jar) to constructor java.security.KeyFactory(java.security.KeyFactorySpi,java.security.Provider,java.lang.String)
WARNING: Please consider reporting this to the maintainers of org.jruby.ext.openssl.SecurityHelper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
OpenSSL::PKey::ECError: Neither PUB key nor PRIV key:
  initialize at org/jruby/ext/openssl/PKeyEC.java:326
      <main> at test.rb:7
```

The key cannot be loaded.

## Test 2: PKey::EC.new on a file containing private key only

Same error on JRuby, MRI works.

## Test 3: PKey.read on a file containing private key only

The key is read successfully but is not sent to the server. s_server side shows:

```
140616554186048:error:1417C0C7:SSL routines:tls_process_client_certificate:peer did not return a certificate:../ssl/statem/statem_srvr.c:3687:
shutting down SSL
CONNECTION CLOSED
```

**Comment by rchekaluk** [comment]
I have found error `OpenSSL::PKey::ECError: Neither PUB key nor PRIV key:` PKeyEC.java:326 to be correlated with https://github.com/jruby/jruby-openssl/issues/189

**Comment by kares** [comment]
expected to be fixed since JOSSL 0.14.4
